corehq/apps/fixtures/upload/run_upload.py

```diff
@@ -56,43 +56,11 @@ def _update_progress(table_count, item_count, items_in_table):
                 DownloadBase.set_progress(task, processed, 10 * total_tables)
 
         for table_number, table_def in enumerate(type_sheets):
-            tag = table_def.table_id
-            new_data_type = FixtureDataType(
-                domain=domain,
-                is_global=table_def.is_global,
-                tag=tag,
-                fields=table_def.fields,
-                item_attributes=table_def.item_attributes
-            )
-            try:
-                tagged_fdt = FixtureDataType.fixture_tag_exists(domain, tag)
-                if tagged_fdt:
-                    data_type = tagged_fdt
-                # support old usage with 'UID'
-                elif table_def.uid:
-                    data_type = FixtureDataType.get(table_def.uid)
-                else:
-                    data_type = new_data_type
-
-                if replace and data_type != new_data_type:
-                    data_type.recursive_delete(transaction)
-                    data_type = new_data_type
-
-                data_type.fields = table_def.fields
-                data_type.item_attributes = table_def.item_attributes
-                data_type.is_global = table_def.is_global
-                assert data_type.doc_type == FixtureDataType._doc_type
-                if data_type.domain != domain:
-                    data_type = new_data_type
-                    return_val.errors.append(
-                        _("'%(UID)s' is not a valid UID. But the new type is created.")
-                        % {'UID': table_def.uid}
-                    )
-                if table_def.delete:
-                    data_type.recursive_delete(transaction)
-                    continue
-            except (ResourceNotFound, KeyError):
-                data_type = new_data_type
+            data_type, delete, err = _create_data_type(domain, table_def, replace, transaction)
```

Comment:
Nitpick: `err` => `error` (real words are better than abbreviations)

Reply:
Generally agree, though I'm a little softer on that when the only usage is in the next line (like here or in list comprehensions)